plugins/generator-1.21.1/neoforge-1.21.1/gui.definition.yaml

```diff
@@ -15,6 +15,9 @@ global_templates:
     name: "@SRCROOT/@BASEPACKAGEPATH/init/@JavaModNameMenus.java"
   - template: elementinits/screens.java.ftl
     name: "@SRCROOT/@BASEPACKAGEPATH/init/@JavaModNameScreens.java"
+  - template: json/gui_sprite.json.ftl

```

Comment:
Looking at the code, (I think, did not verify by compiling/running MC) 1.21.1 still provides a way to blit just part of the image without having to make a new sprite for this. For the sake of code simplicity and code generator performance, it may make sense to avoid making sprite and just using blit directly like in the 1.20.1 code.

------------

This may be irrelevant due to above comment, but still leaving this here for the future PRs.

For JSON files, you should specify the writer type to be JSON or code like this is generated (+ tests don't validate JSON correctly):

![image](https://github.com/user-attachments/assets/6808bda6-f059-44f8-8462-67c99e47c3a9)